testcases/misc_tests/cca_export_import_test.c

```diff
@@ -1611,6 +1675,288 @@ static CK_RV cca_ecc_export_import_tests(void)
+#endif
+
+        // re-import this cca public Dilithium key token as new public ecc key
+
+        snprintf(label, sizeof(label), "re-imported_dilithium_%s_public_key",
+                 dilithium_variants[i].name);
+        rc = import_ibm_dilithium_publ_key(session, label, publ_opaquekey,
+                                           publ_opaquekeylen, &imp_publ_key);
+        if (rc != CKR_OK) {
+            testcase_fail("import_ibm_dilithium_publ_key on exported cca Dilithium key token failed rc=%s",
+                          p11_get_ckr(rc));
+            goto error;
+        }
+
+        // sign with re-imported private key
+
+        rc = funcs->C_SignInit(session, &mech, imp_priv_key);
+        if (rc != CKR_OK) {
+            testcase_error("C_SignInit() with re-imported priv key failed, rc=%s",
+                           p11_get_ckr(rc));
+            goto error;
+        }
+        msglen = sizeof(msg);
+        siglen = sizeof(sig);
+        rc = funcs->C_Sign(session, msg, msglen, sig, &siglen);
+        if (rc != CKR_OK) {
+            testcase_error("C_Sign() with re-imported priv key failed, rc=%s",
+                           p11_get_ckr(rc));
+            goto error;
+        }
+
+        // verify with original public key
+
+        rc = funcs->C_VerifyInit(session, &mech, publ_key);
+        if (rc != CKR_OK) {
+            testcase_error("C_VerifyInit() rc=%s", p11_get_ckr(rc));
+            goto error;
+        }
+        rc = funcs->C_Verify(session, msg, msglen, sig, siglen);
+        if (rc == CKR_OK) {
+            ;
+        } else if (rc == CKR_SIGNATURE_INVALID ||
+                   rc == CKR_SIGNATURE_LEN_RANGE) {
+            testcase_fail("signature verify on signature generated with re-imported priv key failed, rc=%s",
+                          p11_get_ckr(rc));
+            goto error;
+        } else {
+            testcase_error("C_Verify() on signature generated with re-imported priv key failed, rc=%s",
+                           p11_get_ckr(rc));
+            goto error;
+        }
+
+        // sign with original private key
+
+        rc = funcs->C_SignInit(session, &mech, priv_key);
+        if (rc != CKR_OK) {
+            testcase_error("C_SignInit() rc=%s", p11_get_ckr(rc));
+            goto error;
+        }
+        msglen = sizeof(msg);
+        siglen = sizeof(sig);
+        rc = funcs->C_Sign(session, msg, msglen, sig, &siglen);
+        if (rc != CKR_OK) {
+            testcase_error("C_Sign() rc=%s", p11_get_ckr(rc));
+            goto error;
+        }
+
+        // verify with re-imported public key
+
+        rc = funcs->C_VerifyInit(session, &mech, imp_publ_key);
+        if (rc != CKR_OK) {
+            testcase_error("C_VerifyInit() with re-imported pub key failed, rc=%s",
+                           p11_get_ckr(rc));
+            goto error;
+        }
+        rc = funcs->C_Verify(session, msg, msglen, sig, siglen);
+        if (rc == CKR_OK) {
```

Comment:
Hm, why not a switch for the rc? Would probably be better readable ...

Reply:
Yea, again, there are many other places in that test doing it the same way....
Not sure if I really like to change all of these places in this commit....